code/game/objects/structures/xeno.dm

```diff
@@ -129,6 +129,12 @@
 	. = ..()
 	if(builder)
 		linked_carrier = builder
+	RegisterSignal(src, COMSIG_MOVABLE_SHUTTLE_CRUSH, .proc/shuttle_crush)
+
+/obj/effect/alien/resin/trap/proc/shuttle_crush() //No more trapping shuttles with huggies
+	SIGNAL_HANDLER
+	if(hugger)
+		QDEL_NULL(hugger)
```

Comment:
no if needed

Reply:
the destroy might not be enough

```
/obj/effect/alien/resin/trap/Destroy()
	if(hugger && loc)
		drop_hugger()
	return ..()
```
As this will pop out the hugger.

most other ones do something like this
```
/obj/machinery/autodoc/proc/shuttle_crush()
	SIGNAL_HANDLER
	if(occupant)
		var/mob/living/carbon/human/H = occupant
		go_out()
		H.gib()
```

Reply:
@Hughgent It directly deletes the hugger